tests/form-filling.js

```diff
@@ -0,0 +1,101 @@
+const helper = require('./helper');
+const genericFormPo = require('./page-objects/forms/generic-form.po');
+const answeredQuestions = [];
+
+const getPaths = (obj, prevKey, r = []) => {
+  const results = r;
+  let evalObj = obj;
+  if (Array.isArray(obj)) {
+    evalObj = obj[0];
+  }
+  Object.keys(evalObj).forEach((key) => {
+    let resultKey = '';
+    if(prevKey){
+      resultKey = `${prevKey}/${key}`;
+    } else {
+      resultKey = key;
+    }
+
+    if(key === 'inputs'){
+      return;
+    }
+    if(typeof evalObj[key] !== 'object'){
+      results.push({path: resultKey, value: evalObj[key]});
+    }else {
+      getPaths(evalObj[key], resultKey, results);
+    }
+  });
+  return results;
+};
+
+
+const getVisibleQuestions = async () => {
+  const answers = await element.all(by.css('.current input:not([disabled],.ignore)'));
+  const results = [];
+
+  for(let i = 0; i < answers.length; i++){
```

Comment:
It looks like you're needing to [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) `answers` to `results` ;)

Please rewrite this under that line of thinking with `answers.map ...` There's a good amount of other stuff going on here that can probably be tidied up (and made more readable) by moving it into separate functions.